### ISSUE TITLE
Add support for const types in metadata editor

### ISSUE THREAD

**Issue by kevin-bates** [elyra-ai/elyra#2269]
The runtime types PR (#2263) introduces a `runtime_type` property to each schema within the Runtimes schemaspace.  This property denotes the type of runtime to which instances of the schema correspond and uses a `const` meta-property to hold this value - since the hosting schema is one-to-one with a runtime processor implementation (and, therefore, a type).  Here's the schema entry relative to the `kfp` schema:
```JSON
       "runtime_type": {
          "title": "Runtime Type",
          "description": "The runtime associated with this instance",
          "type": "string",
          "const": "KUBEFLOW_PIPELINES"
        },
```

Left unaddressed, this will present a `Runtime Type` property with an empty edit control.  If any value other than the constant value is entered, the validation for that instance will fail.   Since `runtime_type` is not required, the only "working" approach would be to never enter any value (or know the constant value to enter). Because this is a constant, the user should not need to enter a value or update previously persisted constant values.

For `const` types, the metadata editor should unconditionally set the value of the property to the value stored in the `const` meta-property treat the property as read-only.

The idea that constant-based values should perhaps be hidden is also applicable here.  As a result, it might be good to offer a `uihints` "style" that denotes a particular property be completely hidden from the end-user.

**Comment by ptitzler** [comment]
How about introducing a UI property that identifies special attributes, such as 'hidden' (immutable and not rendered) or 'read-only' (immutable)?

**Comment by kevin-bates** [comment]
I think this a lot of sense and is used in other metadata-based systems I've worked on in the past.

**Comment by lresende** [comment]
Based on the conversation above, do we still need this?

**Comment by kevin-bates** [comment]
Yes, the discussion above is essentially regarding the two forms for how we want to _display_ constant-based properties.  We should still transfer the value reflected in the `const` field to the metadata instance.  That transfer isn't really necessary if the `uihint` were to indicate a 'hidden attribute, but is for 'read-only' and 'visible'.

**Comment by akchinSTC** [comment]
See: https://github.com/elyra-ai/elyra/pull/2263#pullrequestreview-803006967

**Comment by akchinSTC** [comment]
@marthacryan - check to see if this has been resolved earlier release
